nighthawk/test/benchmark_http_client_test.cc

```diff
@@ -0,0 +1,289 @@
+#include <chrono>
+
+#include "ares.h"
+
+#include "gtest/gtest.h"
+
+#include "common/api/api_impl.h"
+#include "common/common/compiler_requirements.h"
+#include "common/common/thread_impl.h"
+#include "common/event/dispatcher_impl.h"
+#include "common/filesystem/filesystem_impl.h"
+#include "common/http/header_map_impl.h"
+#include "common/network/utility.h"
+#include "common/runtime/runtime_impl.h"
+#include "common/stats/isolated_store_impl.h"
+#include "common/thread_local/thread_local_impl.h"
+
+#include "nighthawk/source/client/benchmark_client_impl.h"
+#include "nighthawk/source/common/platform_util_impl.h"
+#include "nighthawk/source/common/rate_limiter_impl.h"
+#include "nighthawk/source/common/sequencer_impl.h"
+#include "nighthawk/source/common/statistic_impl.h"
+
+#include "envoy/thread_local/thread_local.h"
+#include "test/mocks/runtime/mocks.h"
+#include "test/mocks/thread_local/mocks.h"
+
+#include "test/integration/http_integration.h"
+
+#include "test/integration/integration.h"
+#include "test/integration/utility.h"
+#include "test/server/utility.h"
+#include "test/test_common/utility.h"
+
+using namespace std::chrono_literals;
+
+namespace Nighthawk {
+
+std::string lorem_ipsum_config;
+
+class BenchmarkClientTest : public Envoy::BaseIntegrationTest,
+                            public testing::TestWithParam<Envoy::Network::Address::IpVersion> {
+public:
+  BenchmarkClientTest()
+      : Envoy::BaseIntegrationTest(GetParam(), realTime(), lorem_ipsum_config),
+        api_(thread_factory_, store_, timeSystem()), dispatcher_(api_.allocateDispatcher()) {}
+
+  static void SetUpTestCase() {
+    Envoy::Filesystem::InstanceImpl filesystem;
+
+    Envoy::TestEnvironment::setEnvVar("TEST_TMPDIR", Envoy::TestEnvironment::temporaryDirectory(),
```

Comment:
How come we need to do this? Doesn't Bazel set this?

Reply:
No, it's not added for us by Bazel, but I removed this here, and added it to the CI script instead, I think that is what Envoy does.